src/classes/HiFiAudioAPIData.ts

```diff
@@ -32,6 +32,188 @@ export class Point3D {
         this.y = y;
         this.z = z;
     }
+
+    /**
+     * Returns the vector dot product between two Point3Ds.
+     */
+    static dot(a: Point3D, b: Point3D): number {
+        return a.x * b.x + a.y * b.y + a.z * b.z;
+    }
+
+    /**
+     * Returns the vector cross product between two Point3Ds.
+     */
+    static cross(a: Point3D, b: Point3D): Point3D {
+        return new Point3D({x: a.y*b.z - a.z*b.y, y: a.z*b.x - a.x*b.z, z: a.x*b.y - a.y*b.x});
+    }
+
+    /**
+      * Negates this Point3D.
+      */
+    negate() {
+        this.x = -this.x;
+        this.y = -this.y;
+        this.z = -this.z;
+    }
+}
+
+class Matrix3 {
+    a: Point3D;
+    b: Point3D;
+    c: Point3D;
+
+    /**
+     * The values in the Matrix3 are stored in three Point3D rows:
+     *
+     * |  <- a -> |
+     * |  <- b -> |
+     * |  <- c -> |
+     *
+     */
+    //constructor(a: Point3D(1, 0, 0), b: Point3D(0, 1, 0), c: Point3D(0, 0, 1)) {
+    constructor(a: Point3D, b: Point3D, c: Point3D) {
+        this.a = a;
+        this.b = b;
+        this.c = c;
+    }
+
+    /** 
+     * Operate from the **left** on a columnar Point3D on the **right**.
+     * Performs an afine transformation on the Point3D and
+     * returns the transformed Point3D in the destination coordinate system.
+     */
+    transform(point: Point3D): Point3D {
+        let p = new Point3D();
+        p.x = Point3D.dot(this.a, point);
+        p.y = Point3D.dot(this.b, point);
+        p.z = Point3D.dot(this.c, point);
+        return p;
+    }
+
+    /**
+     * Flip the Matrix3 about the diagonal axis.
+     *
+     * | ax  ay  az |       | ax  bx  cx |
+     * | bx  by  bz |  -->  | ay  by  cy |
+     * | cx  cy  cz |       | az  bz  cz |
+     *
+     */
+    transpose() {
+        var temp = this.a.y;
+        this.a.y = this.b.x;
+        this.b.x = temp;
+        temp = this.a.z;
+        this.a.z = this.c.x;
+        this.c.x = temp;
+        temp = this.b.z;
+        this.b.z = this.c.y;
+        this.c.y = temp;
+    }
+
+    /**
+     * Multiply the two matrices: f * g
+     * Returns a new Matrix3 that represents their product.
+     */
+    static multiply(f: Matrix3, g: Matrix3): Matrix3 {
+        let a =  new Point3D({
+            x: f.a.x * g.a.x + f.a.y * g.b.x + f.a.z * g.c.x,
+            y: f.a.x * g.a.y + f.a.y * g.b.y + f.a.z * g.c.y,
+            z: f.a.x * g.a.z + f.a.y * g.b.z + f.a.z * g.c.z});
+        let b = new Point3D({
+            x: f.b.x * g.a.x + f.b.y * g.b.x + f.b.z * g.c.x,
+            y: f.b.x * g.a.y + f.b.y * g.b.y + f.b.z * g.c.y,
+            z: f.b.x * g.a.z + f.b.y * g.b.z + f.b.z * g.c.z});
+        let c = new Point3D({
+            x: f.c.x * g.a.x + f.c.y * g.b.x + f.c.z * g.c.x,
+            y: f.c.x * g.a.y + f.c.y * g.b.y + f.c.z * g.c.y,
+            z: f.c.x * g.a.z + f.c.y * g.b.z + f.c.z * g.c.z});
+
+        return new Matrix3(a, b, c);
+    }
+}
```

Comment:
The **Matrix3** class was needed to do the transform calculations in **HiFiCoordinateFrameUtil**.